### PR TITLE
Fix snapshot build, build against OpenSearch 1.1.

### LINKED ISSUE #116: [BUG] Snapshot build is broken
**Describe the bug**

Coming from https://github.com/opensearch-project/opensearch-build/pull/189, attempting to build a snapshot version of this plugin:

```
./gradlew assemble --no-daemon --refresh-dependencies -DskipTests=true -Dopensearch.version=1.0.0-SNAPSHOT -Dbuild.snapshot=true
```

Fails to find the non-snapshot depenency. It should be looking for the snapshot dependency of OpenSearch.

```
To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/6.7/userguide/gradle_daemon.html.
Daemon will be stopped at the end of the build stopping after processing

> Configure project :
mixed cluster flag: false
plugin in the node: /tmp/tmplvhw34rv/src/test/resources/job-scheduler/opensearch-job-scheduler-1.0.0.0.zip
plugin in the node: /tmp/tmplvhw34rv/build/distributions/opensearch-index-management-1.0.0.0-SNAPSHOT.zip
=======================================
OpenSearch Build Hamster says Hello!
  Gradle Version        : 6.7
  OS Info               : Linux 5.8.0-1041-aws (amd64)
  JDK Version           : 14 (JDK)
  JAVA_HOME             : /usr/lib/jvm/java-14-openjdk-amd64
  Random Testing Seed   : BAC5F2377C0E57FB
  In FIPS 140 mode      : false
=======================================

> Task :ktlint
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jetbrains.kotlin.com.intellij.util.ReflectionUtil (file:/home/ubuntu/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-compiler-embeddable/1.4.31/6451ea797cef544e81f507b0d9959cd97ae09c0/kotlin-compiler-embeddable-1.4.31.jar) to method java.util.ResourceBundle.setParent(java.util.ResourceBundle)
WARNING: Please consider reporting this to the maintainers of org.jetbrains.kotlin.com.intellij.util.ReflectionUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

> Task :compileKotlin FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileKotlin'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not find org.opensearch:opensearch-job-scheduler-spi:1.0.0.0.
     Searched in the following locations:
       - https://jcenter.bintray.com/org/opensearch/opensearch-job-scheduler-spi/1.0.0.0/opensearch-job-scheduler-spi-1.0.0.0.pom
       - file:/home/ubuntu/.m2/repository/org/opensearch/opensearch-job-scheduler-spi/1.0.0.0/opensearch-job-scheduler-spi-1.0.0.0.pom
     Required by:
         project :

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 14s
2 actionable tasks: 2 executed
Traceback (most recent call last):
  File "./bundle-workflow/python/build.py", line 56, in <module>
    builder.build(manifest.build.version, arch, args.snapshot)
  File "/home/ubuntu/source/opensearch-project/opensearch-build/dblock-opensearch-build/bundle-workflow/python/build_workflow/builder.py", line 39, in build
    self.git_repo.execute(build_command)
  File "/home/ubuntu/source/opensearch-project/opensearch-build/dblock-opensearch-build/bundle-workflow/python/build_workflow/git_repository.py", line 30, in execute
    subprocess.check_call(command, cwd = self.dir.name, shell = True)
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '/home/ubuntu/source/opensearch-project/opensearch-build/dblock-opensearch-build/bundle-workflow/python/../scripts/bundle-build/standard-gradle-build/build.sh -v 1.0.0 -a x64 -s true -o artifacts' returned non-zero exit status 1.
```

**Expected behavior**
Build to pickup snapshot version of OpenSearch and jobscheduler/spi.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

Fixes #116 and #112. 

Builds job-scheduler and notifications in-line instead of having to commit plugin artifacts.

*CheckList:*
[x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/index-management/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dbbaughe** [comment]
I'd like to keep the test resources checked in for development.
CI can still overwrite them and run the PR tests w/ the latest up to date commits from the dependent plugins.

Trying to limit all of the extra hacky work needed for someone pulling down a plugin. Yes, the checked in plugin is stale compared to potentially what's on the branch in the dependent plugin repo, but even if they check that out and build locally it immediately becomes "stale" too and they would have to always make sure they rebuild to have the "latest" whenever there are changes in the other repositories.

File: .github/workflows/multi-node-test-workflow.yml [hunk 1]
```diff
@@ -39,32 +39,46 @@ jobs:
...
-          repository: 'opensearch-project/alerting'
-          path: alerting
-          ref: '1.0'
-      - name: Build alerting
-        working-directory: ./alerting
-        run: ./gradlew :alerting-notification:publishToMavenLocal -Dopensearch.version=1.0.0 -Dbuild.snapshot=false
-      # index-management
-      - name: Checkout Branch
-        uses: actions/checkout@v2
+          repository: 'dblock/notifications'
```
Comment (dbbaughe) [inline .github/workflows/multi-node-test-workflow.yml:67]: Was this leftover for testing? Can we change this back to the official repo?

File: .github/workflows/test-and-build-workflow.yml [hunk 2]
```diff
@@ -39,32 +39,47 @@ jobs:
...
-          repository: 'opensearch-project/alerting'
-          path: alerting
-          ref: '1.0'
-      - name: Build alerting
-        working-directory: ./alerting
-        run: ./gradlew :alerting-notification:publishToMavenLocal -Dopensearch.version=1.0.0 -Dbuild.snapshot=false
-      # index-management
-      - name: Checkout Branch
-        uses: actions/checkout@v2
+          repository: 'dblock/notifications'
```
Comment (dbbaughe) [inline .github/workflows/test-and-build-workflow.yml:67]: Same comment as above

**Comment by dblock** [comment]
Replaced by https://github.com/opensearch-project/index-management/pull/121

**Pull request closed without merging**